doc/memos/rdm-draft-chrysn-boardnames.md

```diff
@@ -0,0 +1,161 @@
+
+The friendly and the technical name are for boards are derived
+from a designation assigned by the vendor of the defining representative.
+They are composed from the vendor's name and the name chosen by the vendor,
+unless the vendor uses a well known prefix for the board.
+
+When vendor names contain components that are common in names but uncommon in use,
+these are removed.
+For example, "Microchip Technology Inc." is rendered as "Microchip",
+Vendors that primarily use a particular shortened form of their name
+are addressed by that name ("MikroElektronika d.o.o." uses "MIKROE").
+
+As vendors commonly use multiple names for themselves or products
+(eg. "Thunderboard", "Thunderboard Sense" and "SLTB001A"),
+it is up to the board's maintainer to select pick a name
+that is likely to be usable for a long term.
+Marketing names are often unsuitable,
+as vendors (such as "Galaxy A7") might release later revisions without a clear name distinction.
+
+For the **friendly name**,
+the vendor's name and the product name are taken verbatim from the vendor,
+including stylization and capitalization, as far as that is possible within Unicode:
+Examples are "Arduino", "PHYTEC", "micro:bit" and "TRÅDFRI".
+
+A list of vendor names, along with their well known prefixes,
+is maintained as part of the RIOT documentation.
+
+For the **technical name**,
+the friendly names are reduced to the set of lowercase ASCII characters, digits and the dash (`-`) character.
+As a general rule, whitespace in the name is replaced with a dash.
+The precise conversion is left to the maintainer;
+they are encouraged to consider the source language's and ecosystem's conventions.
+For example, "TRÅDFRI" becomes "tradfri"
+and "micro:bit" becomes "microbit"
+(both because that's what the respective vendors use in their URIs).
+
+The vendor and product name are concatenated
+by a space (` `) to form the friendly name,
+and by a dash (`-`) to form the technical name.
+If the product name uses a prefix out of the well known prefix list,
+the vendor name and concatenating character are left out from both names.
+In the context of the vendor,
+the friendly name may be used without the vendor part,
+for example when listing all boards from one vendor.
+
+Note that names can not be reliably split into the vendor and product name;
+such a split can only be performed through inspecting metadata of the board,
+if such are defined.
+
+## Unstructured vendors
+
+Some boards do not have a clear vendor,
+but just "pop up" on online stores.
+Examples are the "bluepill" series,
+and the boards [documented as "nRF52 DK" at the time of writing](https://web.archive.org/web/20211015201641/https://doc.riot-os.org/group__boards__nrf52dk.html).
+
+Unless and until a well-structured vendor of a defining representative is found,
+these are filed under the "No-name" (technical name: "noname") vendor.
+The product name is assigned by the maintainer based on any common term for the boards.
```

Comment:
That's the only bit I'm a bit conflicted about.
Usually those boards *do* have a vendor that produces them, but they might just not be very well known.

E.g. with the classic bluepill boards, that's what they were known as. Slapping a `noname-` prefix would not have helped here at all.

And there are other boards from known vendors, e.g. `samr21-xpro` that don't contain the vendor name. Should that be `microchip-samr21-xpro`?  (And back when it was merged it would have been `atmel-samr21-xpro`

Reply:
Can you find the vendor that produces the black- and bluepill boards? I couldn't, so far.

For the samr series, there may be a useful prefix -- SAM? ATSAM? I don't know these well enough, what's their naming scheme?

Vendor acquisitions are an interesting issue I have not considered, but I'd still rather use aliases here than hope that nobody ever produces collisions, *especially* when boards are named by vendors in an easy-to-confuse way, such as the "nRF52840 MDK USB Dongle".

Reply:
Quoting some conversation with Ben I deem relevant from matrix:

> B: If I understand the proposal correctly it would result in a big rename of e.g. nucleo-f030r8 -> stm-nucleo-f030r8
> C: New boards would be named like that, and documentation can start using consistent names without starting anything destructive. I wouldn't start renaming any board just-because-rdm, at least not unless we have an alias mechanism in place that makes that rename compatible.
> B: Are new nucleo boards then grandfathered in? 
> C: If it helps consistency within that board family, I don't see a reason not to. [...] for large groups of comaintained boards it probably makes sense to take the decision between purity and practicality in a different direction.

Reply:
And before I miss that aspect over the topic of going-on-with-nucleo-boards: assuming we find no vendor for the black/bluepill boards, or the same situation arises with a different board (I don't have a concrete example at hand, but not for lack of off-brand boards, but because those I can find details on quickly all happen to be compatible clones of better-described boards), how would you name them without implying that we know how to characterize that board?

Reply:
I mean currently we just leave it out, but if that's no longer an option due to consistency, I guess `noname` or `generic` are fine.

Reply:
`generic` doesn't sound so judgmental as `noname`. Therefore, I think `generic` is better.

Reply:
It's less judgemental but has has implications of universality -- as if some step of abstraction (generalization) has happened. `unknown` would be accurate but harsh (and hey, it's not our lack of trying!). Maybe `various`? (as in "[Various Artists](https://musicbrainz.org/artist/4e46dd54-81a6-4a75-a666-d0e447861e3f)"; if it does turn out there is a single one we could still alias/rename, as we'd do if the producer of The Blackboard just stepped forward). Or `genericbrand` / `offbrand` or `acme` based on [the Wikipedia article on that topic](https://en.wikipedia.org/wiki/Generic_brand)?

Reply:
Let's just go with "generic".